src/OrchardCore.Modules/OrchardCore.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -391,19 +391,23 @@ private async Task GenerateContainerPathFromPattern(AutoroutePart part)
                     ContentItem = part.ContentItem
                 };
 
-                // We keep the current culture as a reference for later
-                var currentCulture = CultureInfo.CurrentUICulture;
-
                 _contentManager ??= _serviceProvider.GetRequiredService<IContentManager>();
                 var cultureAspect = await _contentManager.PopulateAspectAsync(part.ContentItem, new CultureAspect());
 
-                CultureInfo.CurrentUICulture = cultureAspect.Culture;
-
-                part.Path = await _liquidTemplateManager.RenderAsync(pattern, NullEncoder.Default, model,
-                    scope => scope.SetValue("ContentItem", model.ContentItem));
+                // We keep the current culture as a reference for later
+                var currentCulture = CultureInfo.CurrentUICulture;
 
-                // We reassign the proper culture to the current execution flow
-                CultureInfo.CurrentUICulture = currentCulture;
+                CultureInfo.CurrentUICulture = cultureAspect.Culture;
+                try
```

Comment:
Can I add the try block im PR or you could add it after merge my PR

Reply:
If we use your PR with an using block we would not have to use a try / catch

Reply:
That's why I think before, but what will happen if an exception occurs during the rendering process?

BTW my PR is ready for review

Reply:
Your dispose() will be still called, an using block uses try / finally blocks internally